src/PlaywrightSharp.Tests/Page/Events/PageEventsPopupTests.cs

```diff
@@ -0,0 +1,146 @@
+using System.Threading.Tasks;
+using PlaywrightSharp.Tests.Attributes;
+using PlaywrightSharp.Tests.BaseTests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace PlaywrightSharp.Tests.Page.Events
+{
+    ///<playwright-file>page.spec.js</playwright-file>
+    ///<playwright-describe>Page.Events.Popup</playwright-describe>
+    public class PageEventsPopupTests : PlaywrightSharpPageBaseTest
+    {
+        internal PageEventsPopupTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        ///<playwright-file>page.spec.js</playwright-file>
+        ///<playwright-describe>Page.Events.Popup</playwright-describe>
+        ///<playwright-it>should work</playwright-it>
+        [Fact]
+        public async Task ShouldWork()
+        {
+            var popupTask = Page.OnceAsync<PopupEventArgs>(PageEvent.Popup);

```

Comment:
See this pattern to implement "once"s https://github.com/kblok/puppeteer-sharp/blob/74488cf1760b2d4895a4855e9be87cb50ec3c9d8/lib/PuppeteerSharp.Tests/NetworkTests/SetRequestInterceptionTests.cs#L151

Reply:
I was thinking of adding it since we added `WaitForEvent` and they both work in a similar way and use the same enum

Reply:
`once` is part of the `EventEmitter` class from node, like `on`.  Which are our events.

I think that here https://github.com/microsoft/playwright/blob/v0.10.0/test/page.spec.js#L128 they use `once` as a `WaitForEvent`. We can replace that `once` with a `WaitForEvent`.

But here https://github.com/microsoft/playwright/blob/v0.10.0/test/page.spec.js#L212 they use once by the book. once attaches to the event and deataches after the first event trigger.

So, it's not part of the API, but we can use `WaitForEvent` when then wrap once inside a promise.

Reply:
done